design/userdefined-rootdefinehints.md

```diff
@@ -0,0 +1,148 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# userdefined-rootdevicehints
+
+## Status
+
+provisional
+
+## Table of Contents
+
+<!--ts-->
+   * [userdefined-rootdevicehints](#userdefined-rootdevicehints)
+      * [Status](#status)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+         * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [References](#references)
+
+<!-- Added by: mikkosest -->
+
+<!--te-->
+
+## Summary
+
+This document explains how rootdevicehints could be given as user defined parameters as part BaremetalHostSpec. The idea is that user would have a possibilty to defines selective constrains for root device selection. Sometimes selective constrains will not be completely deterministic, but it is upto user, which hints to use. 
+In legacy, only way to how user can select root device hints used, is to choose the name of the hardware profile which has hardcoded root device hint values. When exposing root device hints as part of the BMH spec, this will be part of deprication steps for hardware profile.
```

Comment:
Originally the only way for an user to select the used root device hints, was to ... 
deprication/deprecation

Reply:
Fixed